src/ClientResolver.php

```diff
@@ -1116,6 +1130,64 @@ public static function _default_signing_region(array &$args)
             : $args['region'];
     }
 
+    public static function _apply_ignore_configured_endpoint_urls($value, array &$args)
+    {
+        $args['config']['ignore_configured_endpoint_urls'] = $value;
+    }
+
+    public static function _default_ignore_configured_endpoint_urls(array &$args)
+    {
+        return ConfigurationResolver::resolve(
+            'ignore_configured_endpoint_urls',
+            false,
+            'bool',
+            $args
+        );
+    }
+
+    public static function _apply_endpoint_url($value, array &$args)
+    {
+        if (empty($value)) {
+            return;
+        }
+
+        $args['config']['endpoint_url'] = $value;
+    }
+
+    public static function _default_endpoint_url(array &$args)
+    {
+        if ($args['config']['ignore_configured_endpoint_urls']
+            || !self::isValidService($args['service'])
```

Comment:
This line is just for tests?

Reply:
More or less. There are other places in the clientresolver where we don't require a service or set a default value when a service is not valid/present.